src/effects.ts

```diff
@@ -0,0 +1,203 @@
+import type { AScene } from "aframe";
+import {
+  BlendFunction,
+  BloomEffect,
+  CopyPass,
+  EffectComposer,
+  EffectPass,
+  LambdaPass,
+  RenderPass,
+  SMAAEffect,
+  ToneMappingEffect,
+  ToneMappingMode
+} from "postprocessing";
+import { Camera, HalfFloatType, Mesh, OrthographicCamera, Scene, WebGLRenderer } from "three";
+import { Layers } from "./camera-layers";
+import { createImageMesh } from "./utils/create-image-mesh";
+import Store from "./storage/store";
+import qsTruthy from "./utils/qs_truthy";
+import ImageMessage from "./react-components/image-message";
+
+export enum AAModes {
+  NONE = "NONE",
+  SMAA = "SMAA",
+  MSAA_2X = "MASS_2X",
+  MSAA_4X = "MASS_4X",
+  MSAA_8X = "MASS_8X"
+}
+
+export function createEffectsComposer(
+  canvas: HTMLCanvasElement,
+  renderer: WebGLRenderer,
+  camera: Camera,
+  scene: Scene,
+  sceneEl: AScene,
+  store: Store
+) {
+  const composer = new EffectComposer(renderer, {
+    frameBufferType: HalfFloatType
+  });
```

Comment:
The core post-processing logic is in an external library `postprocessing`. I don't think I have time to check the lib internal so want to check if it works fine on the test after landing the post-processing feature.

Reply:
Yep. Its definitely worth us reading through and understanding the internals well. I have read a lot of it myself, particularly for the effects we are using.